src/gui/transferlistwidget.cpp

```diff
@@ -59,7 +61,125 @@
 #include "transferlistsortmodel.h"
 #include "updownratiodlg.h"
 
-static QStringList extractHashes(const QList<BitTorrent::TorrentHandle *> &torrents);
+namespace
+{
+    static constexpr int menuHMargin = 3;
+    static constexpr int menuVMargin = 1;
+
+    using ToggleFn = std::function<void (Qt::CheckState)>;
+
+    QStringList extractHashes(const QList<BitTorrent::TorrentHandle *> &torrents)
+    {
+        QStringList hashes;
+        foreach (BitTorrent::TorrentHandle *const torrent, torrents)
+            hashes << torrent->hash();
+
+        return hashes;
+    }
+
+    // Helper for setting style parameters when painting check box primitives.
+    class CheckBoxIconHelper: public QCheckBox
+    {
+    public:
+        explicit CheckBoxIconHelper(QWidget *parent);
+        QSize sizeHint() const override;
+        void initStyleOption(QStyleOptionButton *opt) const;
+
+    protected:
+        void paintEvent(QPaintEvent *) override {}
```

Comment:
I don't want it to paint itself, but I still want it to populate the style options and interact with the layout as if it is visible.